app/uk/gov/hmrc/apipublisher/wiring/ApiPublisher.scala

```diff
@@ -25,10 +25,9 @@ import scala.concurrent.ExecutionContext
 @Singleton
 class ApiPublisher @Inject()(app: Application, registrationService: RegistrationService)(implicit val ec: ExecutionContext) {
 
-    Logger.info(s"Starting api-publisher in mode : ${app.mode}")
-    registrationService.registerPublishCallback().map {
-      _ => Logger.info(s"Publisher has subscribed to the service locator")
-    }.recover {
-      case e => Logger.error(s"Publisher could not subscribe to the service locator", e)
-    }
+  Logger.info(s"Starting api-publisher in mode : ${app.mode}")
+
+  registrationService.register()
+
+  registrationService.subscribeToPublishCallback()
```

Comment:
I'm not sure doing the registration and the subscription in parallel is going to work in dev, where we only have one instance of the microservice. It could happen that it makes the call to register with the service locator, but the service locator won't call the api-publisher as it may not be subscribed for the callbacks yet.

I'm going to merge it anyway as it wouldn't break any existing functionality, and we'll see if we have to tweak it.